<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Automated phylogenetic sequence cluster identification from GenBank • phylotaR</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Automated phylogenetic sequence cluster identification from GenBank">
<meta property="og:description" content="phylotaR is a pipeline for the identification within taxonomic
    groups of orthologous sequence clusters from GenBank as the first step in a
    phylogenetic analysis. The pipeline depends on a local alignment search tool
    and is, therefore, not dependent on differences in gene naming conventions
    and naming errors.">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">phylotaR</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/phylotaR.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AntonelliLab/phylotaR#readme">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    
<p><img src="logo.png" height="300" align="left"></p>
<div id="automated-retrieval-of-orthologous-dna-sequences-from-genbank" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#automated-retrieval-of-orthologous-dna-sequences-from-genbank" class="anchor"></a>Automated Retrieval of Orthologous DNA Sequences from GenBank</h1></div>
<p><a href="https://travis-ci.org/AntonelliLab/phylotaR"><img src="https://travis-ci.org/AntonelliLab/phylotaR.svg?branch=master" alt="Build Status"></a> <a href="https://coveralls.io/github/AntonelliLab/phylotaR?branch=master"><img src="https://coveralls.io/repos/github/AntonelliLab/phylotaR/badge.svg?branch=master" alt="Coverage Status"></a> <a href="https://github.com/ropensci/onboarding/issues/187"><img src="https://badges.ropensci.org/187_status.svg"></a></p>
<p>R implementation of the <a href="http://phylota.net/">PhyLoTa sequence cluster pipeline</a>. For more information see the accompanying website. Tested and demonstrated on Unix and Windows. <strong>Find out more by visiting the <a href="https://antonellilab.github.io/phylotaR/">phylotaR website</a>.</strong></p>
<div id="install" class="section level2">
<h2 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h2>
<p>Currently only the devlopment package is available:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="dt">repo=</span><span class="st">'AntonelliLab/phylotaR'</span>, <span class="dt">build_vignettes=</span><span class="ot">TRUE</span>)</code></pre></div>
<p><strong>Full functionality depends on a local copy of BLAST+ (&gt;= 2.0.0)</strong>. For details on downloading and compiling BLAST+ on your machine please visit the <a href="https://www.ncbi.nlm.nih.gov/books/NBK279690/">NCBI website</a>.</p>
</div>
<div id="pipeline" class="section level2">
<h2 class="hasAnchor">
<a href="#pipeline" class="anchor"></a>Pipeline</h2>
<p><code>phylotaR</code> runs the PhyLoTa pipeline in four automated stages: identify and retrieve taxonomic information on all descendent nodes of the taxonomic group of interest (<code>taxise</code>), download sequence data for every identified node (<code>download</code>), identify orthologous clusters using BLAST (<code>cluster</code>), and identify sister clusters for sets of clusters identified in the previous stage (<code>cluster^2</code>) After these stages are complete, <code>phylotaR</code> provides tools for exploring, identifying and exporting suitable clusters for subsequent analysis.</p>
<p><img src="https://raw.githubusercontent.com/AntonelliLab/phylotaR/master/other/stages.png" alt="phylotaR pipeline"></p>
<p>For more information on the pipeline and how it works see the publication, <a href="https://doi.org/10.3390/life8020020">phylotaR: An Automated Pipeline for Retrieving Orthologous DNA Sequences from GenBank in R</a>.</p>
</div>
<div id="running" class="section level2">
<h2 class="hasAnchor">
<a href="#running" class="anchor"></a>Running</h2>
<p>At a minimum all a user need do is provide the taxonomic ID of their chosen taxonomic group of interest. For example, if you were interested in primates, you can visit the <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/">NCBI taxonomy home page</a> and search primates to look up their ID. After identifying the ID, the <code>phylotaR</code> pipeline can be run with the following script.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phylotaR)
wd &lt;-<span class="st"> '[FILEPATH TO WORKING DIRECTORY]'</span>
ncbi_dr &lt;-<span class="st"> '[FILEPATH TO COMPILED BLAST+ TOOLS]'</span>
txid &lt;-<span class="st"> </span><span class="dv">9443</span>  <span class="co"># primates ID</span>
<span class="kw"><a href="reference/setup.html">setup</a></span>(<span class="dt">wd =</span> wd, <span class="dt">txid =</span> txid, <span class="dt">ncbi_dr =</span> ncbi_dr)
<span class="kw"><a href="reference/run.html">run</a></span>(<span class="dt">wd =</span> wd)</code></pre></div>
<p>The pipeline can be stopped and restarted at any point without loss of data. For more details on this script, how to change parameters, check the log and details of the pipeline, please check out the package vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(phylotaR)
<span class="kw">vignette</span>(<span class="st">"phylotaR"</span>)</code></pre></div>
</div>
<div id="timings" class="section level2">
<h2 class="hasAnchor">
<a href="#timings" class="anchor"></a>Timings</h2>
<p>How long does it take for a phylotaR pipeline to complete? Below is a table listing the runtimes in minutes for different demonstration, taxonomic groups.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Taxon</th>
<th align="right">Taxa</th>
<th align="right">Sequences</th>
<th align="right">Clusters</th>
<th align="right">Taxise</th>
<th align="right">Download</th>
<th align="right">Cluster</th>
<th align="right">Cluster2</th>
<th align="right">Total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Anisoptera</td>
<td align="right">1143</td>
<td align="right">10619</td>
<td align="right">784</td>
<td align="right">0.75</td>
<td align="right">14</td>
<td align="right">14</td>
<td align="right">0.017</td>
<td align="right">28</td>
</tr>
<tr class="even">
<td align="left">Acipenseridae</td>
<td align="right">51</td>
<td align="right">2417</td>
<td align="right">332</td>
<td align="right">0.067</td>
<td align="right">35</td>
<td align="right">1.5</td>
<td align="right">0</td>
<td align="right">37</td>
</tr>
<tr class="odd">
<td align="left">Tinamiformes</td>
<td align="right">25</td>
<td align="right">250</td>
<td align="right">98</td>
<td align="right">0.05</td>
<td align="right">1.9</td>
<td align="right">0.083</td>
<td align="right">0</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Aotus</td>
<td align="right">13</td>
<td align="right">1499</td>
<td align="right">193</td>
<td align="right">0.05</td>
<td align="right">2</td>
<td align="right">0.12</td>
<td align="right">0</td>
<td align="right">2.1</td>
</tr>
<tr class="odd">
<td align="left">Bromeliaceae</td>
<td align="right">1172</td>
<td align="right">9840</td>
<td align="right">721</td>
<td align="right">0.7</td>
<td align="right">18</td>
<td align="right">9.4</td>
<td align="right">0.017</td>
<td align="right">28</td>
</tr>
<tr class="even">
<td align="left">Cycadidae</td>
<td align="right">353</td>
<td align="right">8331</td>
<td align="right">540</td>
<td align="right">0.18</td>
<td align="right">13</td>
<td align="right">3.4</td>
<td align="right">0</td>
<td align="right">17</td>
</tr>
<tr class="odd">
<td align="left">Eutardigrada</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">0.28</td>
<td align="right">5</td>
<td align="right">0.82</td>
<td align="right">0</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">Kazachstania</td>
<td align="right">26</td>
<td align="right">164</td>
<td align="right">33</td>
<td align="right">0.067</td>
<td align="right">9.3</td>
<td align="right">1.1</td>
<td align="right">0.033</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">Platyrrhini</td>
<td align="right">273</td>
<td align="right">19384</td>
<td align="right">543</td>
<td align="right">0.2</td>
<td align="right">30</td>
<td align="right">1.6</td>
<td align="right">0.28</td>
<td align="right">32</td>
</tr>
</tbody>
</table>
<p>To run these same demonstrations see <a href="https://github.com/AntonelliLab/phylotaR/blob/master/demos/demo_run.R">´demos/demo_run.R´</a>.</p>
</div>
<div id="license" class="section level2">
<h2 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h2>
<p>MIT</p>
</div>
<div id="version" class="section level2">
<h2 class="hasAnchor">
<a href="#version" class="anchor"></a>Version</h2>
<p>Version 1.</p>
</div>
<div id="authors" class="section level2">
<h2 class="hasAnchor">
<a href="#authors" class="anchor"></a>Authors</h2>
<p>Dom Bennett (maintainer, R package dev), Hannes Hettling (workhouse code dev), Rutger Vos, Alexander Zizka and Alexandre Antonelli</p>
</div>
<div id="reference" class="section level2">
<h2 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h2>
<p>Bennett, D., Hettling, H., Silvestro, D., Zizka, A., Bacon, C., Faurby, S., … Antonelli, A. (2018). phylotaR: An Automated Pipeline for Retrieving Orthologous DNA Sequences from GenBank in R. <em>Life</em>, <strong>8</strong>(2), 20. <a href="https://doi.org/10.3390/life8020020">DOI:10.3390/life8020020</a></p>
<p>Sanderson, M. J., Boss, D., Chen, D., Cranston, K. A., &amp; Wehe, A. (2008). The PhyLoTA Browser: Processing GenBank for molecular phylogenetics research. <em>Systematic Biology</em>, <strong>57</strong>(3), 335–346. <a href="https://doi.org/10.1080/10635150802158688">DOI:10.1080/10635150802158688</a></p>
</div>
</div>

  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/AntonelliLab/phylotaR#readme">https://​github.com/​AntonelliLab/​phylotaR#readme</a>
</li>
<li>Report a bug at <br><a href="https://github.com/AntonelliLab/phylotaR/issues">https://​github.com/​AntonelliLab/​phylotaR/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Hannes Hettling <br><small class="roles"> Author </small>  </li>
<li>Rutger Vos <br><small class="roles"> Author </small>  </li>
<li>Alexander Zizka <br><small class="roles"> Author </small>  </li>
<li>Dom Bennett <br><small class="roles"> Author, maintainer </small>  </li>
<li>Alexandre Antonelli <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Hannes Hettling, Rutger Vos, Alexander Zizka, Dom Bennett, Alexandre Antonelli.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
